core-bundle/src/Security/Authentication/Token/TokenChecker.php

```diff
@@ -174,7 +174,16 @@ private function getTokenFromStorage(string $context): ?TokenInterface
 
     private function getTokenFromSession(string $sessionKey): ?TokenInterface
     {
-        if (!$this->session->isStarted() || !$this->session->has($sessionKey)) {
+        if (!$this->session->isStarted()) {
+            $request = $this->requestStack->getMasterRequest();
+
+            if (!$request || !$request->hasPreviousSession()) {
+                return null;
+            }
+        }
+
+        // This will start the session if Request::hasPreviousSession() was true
+        if (!$this->session->has($sessionKey)) {
```

Comment:
I don't like trusting on the internal be behaviour of the session component. I prefer starting the session explicitly. The behavior is not documented in the `SessionInterface`.

Reply:
Have you ever started the session manually before accessing it? I don't think anyone does, the only check I ever see is **not** accessing the session if it hasn't been started (because accessing it would start it).

Reply:
Yes, I have seen it before (e.g. https://github.com/m-vo/contao-survey/pull/31/files#diff-51927511979faf55d65c36a68217dc98328ead09b2fdca0d71b57bf8ef3b4d72R248) and usually do it myself. I started doing such stuff which looks not necessary at the first time because the relying on internal implementations causes bugs when things change in the future. Maybe they will, but if they do, I don't have to care. But okay, it's not my code base, just my opionion. :-)

Reply:
> The behavior is not documented in the `SessionInterface`.

It is documented [here](https://symfony.com/doc/current/session.html#avoid-starting-sessions-for-anonymous-users):

> Sessions are automatically started whenever you read, write or even check for the existence of data in the session.

Reply:
And still there are edge cases: In the referenced example where we're reading/writing into a custom attribute bag the session is not started for instance.

Reply:
Then Symfony either needs to fix that or update its documentation :). Can you provide a PR? 😁

Reply:
Maybe, yes. :smile: But on the other hand you're technically not dealing with the `SessionInterface` directly in this case.